src/relstorage/adapters/mysql/adapter.py

```diff
@@ -314,31 +319,9 @@ def _best_lock_objects_and_detect_conflicts(self, cursor, read_current_oids):
         proc_result = multi_results.pop()
         assert not proc_result, proc_result
 
-        # With read_current_oids, the proc returns one or two results,
```

Comment:
I believe I can restore these asserts. I removed them because I saw some failures during an environment restart, and I assumed it had to do with the database shutting down or being overloaded or such. But I now believe it happens when an 3.0a8 server is running against a set of 3.0a7 stored procs, as would happen during a rolling up/downgrade.

Long term we'll need a better strategy to make rolling upgrades possible, or completely forbidden.